src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -91,7 +93,7 @@ public Claim getClaimByLetterHolderId(String id, String authorisation) {
             .orElseThrow(() -> new NotFoundException("Claim not found for letter holder id " + id));
     }
 
-    public Claim getClaimByExternalId(String externalId, String authorisation) {
+    public Claim getClaimByExternalId(String externalId, String authorisation, CaseState caseState) {
```

Comment:
Why has this parameter been added? It seems to be set to open everywhere. Being open or on hold shouldn’t be a concern at this level

Reply:
As far as I know when case is settled it changes state to 'closed' which:
1) breaks the app as backend is unable to retrieve closed case before producing event
2) removes closed cases from any data retrieval that will result settled case not be displayed on the dashboard / available to users anymore

@dharmendrak as far as I understand fixes #1 but not #2. I find this solution incorrect.

Reply:
it was left in error, was not used